tee-worker/ts-tests/batch.test.ts

```diff
@@ -0,0 +1,127 @@
+                }
+            });
+
+        const events = await listenEvent(context.substrate, 'identityManagement', ['IdentityCreated', 'IdentityCreated']);
+        expect(events.length).to.be.equal(2);
+    });
+
+    step('batch test: verify identity', async function () {
+        // Verify Identity: twitter 1
+        const identity_encode = context.substrate.createType('LitentryIdentity', twitterIdentity).toHex();
+        const validation_encode = context.substrate.createType('LitentryValidationData', twitterValidationData).toHex();
+        const identity_ciphertext = encryptWithTeeShieldingKey(context.teeShieldingKey, identity_encode).toString('hex');
+        const validation_ciphertext = encryptWithTeeShieldingKey(context.teeShieldingKey, validation_encode).toString('hex');
+        const verify_tx = context.substrate.tx.identityManagement.verifyIdentity(
+            context.shard,
+            `0x${identity_ciphertext}`,
+            `0x${validation_ciphertext}`
+        );
+
+        // Construct the batch and send the transactions
+        const txs = [verify_tx];
+        await context.substrate.tx.utility
+            .batchAll(txs)
+            .signAndSend(context.defaultSigner[0], ({ status }) => {
+                if (status.isInBlock) {
+                    console.log(`included in ${status.asInBlock}`);
+                }
+            });
+
+        const events = await listenEvent(context.substrate, 'identityManagement', ['IdentityVerified']);
+        expect(events.length).to.be.equal(1);
+    });
+
+    step('batch test: remove identities', async function () {
+        // Remove Identity: twitter 1
+        const rm_id_encode = context.substrate.createType('LitentryIdentity', twitterIdentity).toHex();
+        const rm_id_ciphertext = encryptWithTeeShieldingKey(context.teeShieldingKey, rm_id_encode).toString('hex');
+        const rm_id_tx = context.substrate.tx.identityManagement.removeIdentity(context.shard, `0x${rm_id_ciphertext}`);
+        // Remove Identity: twitter 2
+        const rm_id_encode2 = context.substrate.createType('LitentryIdentity', twitterIdentity2).toHex();
+        const rm_id_ciphertext2 = encryptWithTeeShieldingKey(context.teeShieldingKey, rm_id_encode2).toString('hex');
+        const rm_id_tx2 = context.substrate.tx.identityManagement.removeIdentity(context.shard, `0x${rm_id_ciphertext2}`);
+
+
+        // Construct the batch and send the transactions
+        const txs = [rm_id_tx, rm_id_tx2];
+        await context.substrate.tx.utility
+            .batchAll(txs)
+            .signAndSend(context.defaultSigner[0], ({ status }) => {
+                if (status.isInBlock) {
+                    console.log(`included in ${status.asInBlock}`);
+                }
+            });
+
+        const events = await listenEvent(context.substrate, 'identityManagement', ['IdentityRemoved', 'IdentityRemoved']);
+        expect(events.length).to.be.equal(2);
```

Comment:
I suggest some more test cases:
- error case: the `batch` call is not supported, only `batch_all`
- error case: a wrong `create_identity` in between that throws `UnauthorisedUser` error
- unsupported calls: insert some other calls in between (like balance transfer)
- out-of-order calls: to change the order of calls (like first remove, then create) - all calls should be handled correctly.

And more test cases that you can think of - the principle is to test as many "abnormal" cases as possible - the F/E (or evil users) could throw any calls in a batch.

It can be too much for this single PR, we can have another PR for it (if so please open another issue :))

Reply:
Ok. Let's make another PR.